runhouse/resources/hardware/cluster.py

```diff
@@ -1077,12 +1115,15 @@ def run(
             >>> cpu.run(["python script.py"], run_name="my_exp")
             >>> cpu.run(["python script.py"], node="3.89.174.234")
         """
+        if isinstance(commands, str):
+            commands = [commands]
+
         if node == "all":
             res_list = []
             for node in self.ips:
                 res = self.run(
                     commands=commands,
-                    env=env,
+                    env=env or self.default_env,
                     stream_logs=stream_logs,
                     port_forward=port_forward,
                     require_outputs=require_outputs,
```

Comment:
<!--__GRAPHITE_HTML_TAG_START__--><i class='graphite__hidden'>[Re: line 1146]<br/><br/></i>

<!--__GRAPHITE_HTML_TAG_END__-->still have `base_env` here dk if that's intentional<!--__GRAPHITE_HTML_TAG_START__--><span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/run-house/runhouse/678?utm_source=unchanged-line-comment">Graphite</a>.</span>

<!--__GRAPHITE_HTML_TAG_END__-->